verilog/CST/type.cc

```diff
@@ -153,4 +153,15 @@ const verible::SyntaxTreeLeaf& GetSymbolIdentifierFromEnumName(
   return verible::GetSubtreeAsLeaf(enum_name, NodeEnum::kEnumName, 0);
 }
 
+const verible::SyntaxTreeLeaf* GetTypeIdentifierFromDataType(
+    const verible::Symbol& data_type) {
+  const verible::Symbol* identifier =
+      verible::GetSubtreeAsSymbol(data_type, NodeEnum::kDataType, 0);
+  if (identifier == nullptr ||
+      NodeEnum(identifier->Tag().tag) != NodeEnum::kUnqualifiedId) {
+    return nullptr;
+  }
+  return AutoUnwrapIdentifier(*identifier);
```

Comment:
Actually it would be very helpful if `AutoUnwrapIdentifier` returns list of identifiers we already have some issues for this like this case: `pkg::class_type my_class = new()` and any similar case would crash or wouldn't be extracted.

can `AutoUnwrapIdentifier` be modified to work with `kQualifiedId` nodes?

Reply:
If you `FindAllIdentifiers` on a `kQualifiedId`, that would already give you a list of identifiers where you want them (in order too).

Reply:
But better would be a structured walk over anything with general identifier placement.